libs/community/langchain_community/tools/databricks/_execution.py

```diff
@@ -174,13 +180,40 @@ def execute_function(
         parameters=parametrized_statement.parameters,
         **execute_statement_args,  # type: ignore
     )
-    status = response.status
-    assert status is not None, f"Statement execution failed: {response}"
-    if status.state != StatementState.SUCCEEDED:
-        error = status.error
+    if response.status and job_pending(response.status.state) and response.statement_id:
+        statement_id = response.statement_id
+        _logger.info("Retrying to get statement execution status...")
```

Comment:
Can we merge this statement with the counter and add the iteration count + retry delay within the statement to make the informative statement more useful?